stdlib/public/Concurrency/Task.cpp

```diff
@@ -109,12 +109,14 @@ void AsyncTask::completeFuture(AsyncContext *context, ExecutorRef executor) {
   assert(queueHead.getStatus() == Status::Executing);
 
   // If this is task group child, notify the parent group about the completion.
-  if (isTaskGroupChild()) {
+  fprintf(stderr, "[%s:%d] (%s) complete task %d\n", __FILE__, __LINE__, __FUNCTION__, this);
+  if (hasGroupChildFragment()) {
+    fprintf(stderr, "[%s:%d] (%s) complete group child task %d\n", __FILE__, __LINE__, __FUNCTION__, this);
     // then we must offer into the parent group that we completed,
     // so it may `next()` poll completed child tasks in completion order.
-    auto parent = childFragment()->getParent();
-    assert(parent->isTaskGroup());
-    parent->groupOffer(this, context, executor);
+    auto group = groupChildFragment()->getGroup();
+    assert(group);
+    group->offer(this, context, executor);
```

Comment:
This is the reason we create the task group child fragment, so we can store the group pointer and offer to it when we complete it,

Reply:
I wonder if the general child fragment should just have a pointer to the owning group/record instead of directly to the parent.